### PR TITLE
Add KNN and ROC curve

### PR DESCRIPTION
**Pull request by koalive**
The following pull request adds, fixes, and/or modifies:

* Adds a simple KNN classifier (needed for the following point)
* Adds the ROC AUC metric (solving #32)

I performed the following prior to filing this pull request:

- [x] Tested that my change does not break the analysis pipeline
- [x] Ran a linter through my code
- [x] Update environment dependencies if my code introduces a new package

### CODE REVIEW

**Comment by wagenrace** [comment]
Looks great. KNN is a great addition to the research. But was wondering a few things:
- In you README file you refer to results/all_scores.csv but this file is missing from the git
- Also what is it you exactly mean with '0.300 ± 0.010, 10, 15, 25, 35 or 50. ' in the README?

I try to run your code at the moment but this might take some time (partly because I run everything in Windows Subsystem Linux, partly other thing to do on my days off :) )

**Comment by wagenrace** [comment]
Small thing for the next time: split your work in smallest pieces. You have now KNN researched (second pull request) and the functionelity of ROC AUC (first pull request). Makes it easier for use to check, we can approve ROC AUC and add it even is KNN has some work to do

**Comment by koalive** [comment]
Thanks! Just added the results and tried to clarify that. KNN is deterministic so I ran it with 6 different values for the number of neighbors considered to estimate the robustness of the performance. The proper way to do that would be with a proper cross-validation.
You're right, the issue was that the ROC curves needed the KNN (or any probabilistic prediction model) to be tested so I thought bundling them in a single PR was the least inconvenient solution for now. KNN implementation does not need ROC curves so we can probably split it if this is needed. 
Cheers.

**Comment by wagenrace** [comment]
Cool, it means you have parameter estimation in your solution. But with deterministic solutions it becomes a bit more tricky, I am having the same problem with support vector machine (SVM). Common way of solving it is with [cross-validation ](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_validate.html) but that will be out of scope for this pull request. 

I will test if I can run your code now

**Comment by gwaygenomics** [comment]
@wagenrace @koalive do you need me to do anything here? I can try running on my local machine

**Comment by wagenrace** [comment]
@koalive sorry for the late reaction. I missed your updates. Everything look good, I will merge it to the project
